Src/StdLib/Lib/test/test_iterlen.py

```diff
@@ -223,7 +232,12 @@ def test_issue1242657(self):
     def test_invalid_hint(self):
         # Make sure an invalid result doesn't muck-up the works
         self.assertEqual(list(NoneLengthHint()), list(range(10)))
-
+        # It's a TypeError if __length_hint__ returns something other than an int or NotImplemented
+        self.assertRaises(TypeError, list, StringLengthHint())
```

Comment:
We typically put any custom tests in a file under the root `Tests` folder instead of modifying the StdLib.

Reply:
I should have time to pull this out tomorrow.